From owner-tkined@ibr.cs.tu-bs.de Mon Feb 12 18:35:40 1996
Return-Path: <owner-tkined@ibr.cs.tu-bs.de>
Received: from dan.ods.com ([160.86.13.39]) by elo.ods.com
	 with smtp id <m0tm8ia-000BipC@elo.ods.com>
	(Debian /\oo/\ Smail3.1.29.1 #29.36); Mon, 12 Feb 96 18:35 CST
Received: from ra.ibr.cs.tu-bs.de by dan.ods.com (4.1/SMI-4.1)
	id AA11725; Mon, 12 Feb 96 18:32:22 CST
Received: from ares [134.169.34.25] by ra.ibr.cs.tu-bs.de (8.6.10/tubsibr) with ESMTP id VAA10737 for <tkined@ibr.cs.tu-bs.de>; Mon, 12 Feb 1996 21:24:09 +0100
Received: from schoenw@localhost by ares.ibr.cs.tu-bs.de (8.6.10/tubsibr) id VAA18522; Mon, 12 Feb 1996 21:24:09 +0100
Date: Mon, 12 Feb 1996 21:24:09 +0100
Message-Id: <199602122024.VAA18522@ares.ibr.cs.tu-bs.de>
From: Juergen Schoenwaelder <schoenw@ibr.cs.tu-bs.de>
To: tkined@ibr.cs.tu-bs.de
Subject: fix for an event handling bug in scotty-2.0.2
Reply-To: schoenw@ibr.cs.tu-bs.de
Status: RO

Hi!

Oliver Goldman <goldman@alantec.com> reported a bug in scotty-2.0.2
which can cause core dumps if you are processing events (e.g. calling
update) from a callback of an asynchronous SNMP request. Below is a
fix that should cure this problem.

							Juergen

*** snmpUtil.c-orig     Fri Feb  9 20:15:37 1996
--- snmpUtil.c  Fri Feb  9 20:15:16 1996
***************
*** 748,753 ****
--- 748,756 ----
      for (p = session->requestList, q = NULL; p != NULL; q = p,p = p->nextPtr) {
        if (p == request) break;
      }
+ 
+     if (p == NULL) return;
+ 
      if (q == NULL) {
        session->requestList = p->nextPtr;
      } else {

